### ISSUE TITLE
Lessons learned from class iter=1

### ISSUE THREAD

**Issue by polina-tsvilodub** [CogSciPrag/Understanding-LLMs-course#12]
This is the place to collect feedback and thoughts for improving the next iteration of the class.
- order / structure of topics:
    - filling out some of the sessions with more concrete and nice examples / recaps will be helpful
    - in terms of topic order, if possible, it might make sense to cover standard (behvaioral) evals first, then, try to put attribution and mechanistic interpretability blocks together
- the following topics could, in principle, benefit from details on the blackboard (could also be covered in the tutorials, like the transformers example):
    - backprop step
    - maybe aspects of the RL / fine-tuning pipeline
        - e.g., dissecting PPO / at least REINFORCE
    - evaluation metrics, calculation by hand
    - shapley values, gradient tracing example
    - early decoding operation, residual stream step.
- sheets to be improved:
    - sheet 2.2 / 2.3: maybe it should rather be an example of a classifier? (comes in handier later)
    - 3.1: the transformers part wasn’t a great fit with the lecture, it was there because it was already required for the HW.
    - 3.3: fancy prompting requires fancy model
    - 4.1: add explanations of debugging / interpretation of the various things one sees with the libraries
    - 5.1: fancy tools require accounts; add working with memory / examples
    - 6.1: add actual explanations of the different methods of inseq (see blackboard part); add example with Shapley; actually test the classifier beforehand
    - 7.1: add conceptual background materials for the psych part
    - 7.2: spell it out better, relate it to the lecture (should have been the philo lecture)
    - 8.1: add something about the circuits / head analysis
    - fix numbering of sheets
- structure overall:
    - we might rethink the availability of recordings / attendance (it seems that only <0.5 of the people actually attended)
    - feedback on the sheets was that they were partially too long to be completed in the in-class time. Furthermore, there was little incentive to look at them. Maybe there is an efficient way to have some of them actually be mandatory? *although we do have solutions online now)
        -  in general, with sheets / HW in place, it would be good to close the gap between the exam and sheets / homework a bit more
    - I need to do better compute resource management / testing (some homework tasks turned out to be tricky to solve with colab resources)
    - there should be harder deadlines for grading HW (re: exam)
        - some HW tasks should be streamlines / less open-ended to be easier to grade
    - interested students asked about more background materials on:
        - ANNs for linguistics folks
        - exp. design / linguistics for ML folks
        - we might, in general, benefit from a solid stack of materials and work sheets
    - we might think again about the length of sessions, some feedback stated that 3h were quite hard.
    - we might want to sharpen the narrative / the learning goals of the course + tutorial (while keeping in mind the heterogeneous audience we are covering)
